drivers/serial/uart_stm32.c

```diff
@@ -287,15 +287,28 @@ static int uart_stm32_init(struct device *dev)
 			       (clock_control_subsys_t *)&config->pclken,
 			       &clock_rate);
 
-	LL_USART_SetBaudRate(UartInstance,
-			     clock_rate,
+#if defined(CONFIG_SOC_SERIES_STM32L0X) || defined(CONFIG_SOC_SERIES_STM32L4X)
+	if (IS_LPUART_INSTANCE(UartInstance)) {
+		LL_LPUART_SetBaudRate(UartInstance,
+				      clock_rate,
 #ifdef USART_PRESC_PRESCALER
-			     LL_USART_PRESCALER_DIV1,
+				      LL_LPUART_PRESCALER_DIV1,
+#endif
+				      config->baud_rate);
+	} else {
+#else
+	if (1) {
+#endif
+		LL_USART_SetBaudRate(UartInstance,
+				     clock_rate,
+#ifdef USART_PRESC_PRESCALER
+				     LL_USART_PRESCALER_DIV1,
 #endif
 #ifdef USART_CR1_OVER8
-			     LL_USART_OVERSAMPLING_16,
+				     LL_USART_OVERSAMPLING_16,
 #endif
-			     config->baud_rate);
+				     config->baud_rate);
+	}
```

Comment:
This part starts to be really hard to read due to the excessive ifdef'ery. I had to read twice to understand why the `if (1)` in the middle. What about completely decoupling STM32Lxx from the rest as both parts have few in common?
```
#if defined(CONFIG_SOC_SERIES_STM32L0X) || defined(CONFIG_SOC_SERIES_STM32L4X)
    if (IS_LPUART_INSTANCE(UartInstance)) {
        LL_LPUART_SetBaudRate(...);
    }
#else /* CONFIG_SOC_SERIES_STM32Lxx */
    LL_USART_SetBaudRate(...);
#endif
```

Reply:
Well, then heavily ifdef'ed code for ordinary USARTs will be duplicated like that:
```
#if defined(CONFIG_SOC_SERIES_STM32L0X) || defined(CONFIG_SOC_SERIES_STM32L4X)
    if (IS_LPUART_INSTANCE(UartInstance)) {
        LL_LPUART_SetBaudRate(...);
    } else {
        LL_USART_SetBaudRate(...);
    }
#else /* CONFIG_SOC_SERIES_STM32Lxx */
    LL_USART_SetBaudRate(...);
#endif
```
Is that really better?

Reply:
Another option is to add use a new function either in this file, either in new file uart_lx_stm32.c

Reply:
I've tried to make this thing more readable in today's push. Could you look at that?

Reply:
It is far more readable, thank you.